var/spack/repos/builtin/packages/minizip/static.patch

```diff
@@ -0,0 +1,26 @@
+--- a/contrib/minizip/Makefile.am	2012-03-26 22:17:41.000000000 -0500
++++ b/contrib/minizip/Makefile.am	2020-12-26 12:48:31.000000000 -0600
+@@ -8,7 +8,7 @@
+ zlib_top_builddir = $(top_builddir)/../..
+ 
+ AM_CPPFLAGS = -I$(zlib_top_srcdir)
+-AM_LDFLAGS = -L$(zlib_top_builddir)
++AM_LDFLAGS = $(zlib_top_builddir)/libz.a
```

Comment:
Does this require `zlib~shared`? The `zlib` recipe has
```
        if '~shared' in spec:
            config_args.append('--static')
```
so it might not create static libraries otherwise.

Reply:
It's linking to the local build anyway. That's why we build zlib in the root before building minizip in contrib.